gn3/db/datasets.py

```diff
@@ -1,8 +1,57 @@
 """
 This module contains functions relating to specific trait dataset manipulation
 """
+import os
 from typing import Any
 
+def retrieve_sample_list(group: str):
+    """
+    Get the sample list for a group (a category that datasets belong to)
+
+    Currently it is fetched from the .geno files, since that's the only place
+    the "official" sample list is stored
+    """
+
+    genofile_path = os.environ.get("GENENETWORK_FILES") + "/genotype/" + group + ".geno"
+    if os.path.isfile(genofile_path):
+        genofile = open(genofile_path)
+
+        for line in genofile:
+            line = line.strip()
+            if not line:
+                continue
+            if line.startswith(("#", "@")):
+                continue
+            break
+
+        headers = line.split("\t")
+
+        if headers[3] == "Mb":
+            samplelist = headers[4:]
+        else:
+            samplelist = headers[3:]
+        return samplelist
```

Comment:
*issue:* When opening a file, use a context-manager, in this case, `with open ...`.  This way, you don't have to worry about closing any hanging file connections should something unforeseen happens.